src/vm/functions/mod.rs

```diff
@@ -91,86 +91,87 @@ define_named_enum!(NativeFunctions {
-            StxBurn => CallableType::SpecialFunction("special_stx_burn", &assets::special_stx_burn),
+            Add => NativeFunction("native_add", NativeHandle::MoreArg(&arithmetic::native_add), cost_functions::ADD),
+            Subtract => NativeFunction("native_sub", NativeHandle::MoreArg(&arithmetic::native_sub), cost_functions::SUB),
+            Multiply => NativeFunction("native_mul", NativeHandle::MoreArg(&arithmetic::native_mul), cost_functions::MUL),
+            Divide => NativeFunction("native_div", NativeHandle::MoreArg(&arithmetic::native_div), cost_functions::DIV),
+            CmpGeq => NativeFunction("native_geq", NativeHandle::DoubleArg(&arithmetic::native_geq), cost_functions::GEQ),
+            CmpLeq => NativeFunction("native_leq", NativeHandle::DoubleArg(&arithmetic::native_leq), cost_functions::LEQ),
+            CmpLess => NativeFunction("native_le", NativeHandle::DoubleArg(&arithmetic::native_le), cost_functions::LE),
+            CmpGreater => NativeFunction("native_ge", NativeHandle::DoubleArg(&arithmetic::native_ge), cost_functions::GE),
+            ToUInt => NativeFunction("native_to_uint", NativeHandle::SingleArg(&arithmetic::native_to_uint), cost_functions::INT_CAST),
+            ToInt => NativeFunction("native_to_int", NativeHandle::SingleArg(&arithmetic::native_to_int), cost_functions::INT_CAST),
+            Modulo => NativeFunction("native_mod", NativeHandle::DoubleArg(&arithmetic::native_mod), cost_functions::MOD),
+            Power => NativeFunction("native_pow", NativeHandle::DoubleArg(&arithmetic::native_pow), cost_functions::POW),
+            BitwiseXOR => NativeFunction("native_xor", NativeHandle::DoubleArg(&arithmetic::native_xor), cost_functions::XOR),
+            And => SpecialFunction("native_and", &boolean::special_and),
+            Or => SpecialFunction("native_or", &boolean::special_or),
+            Not => NativeFunction("native_not", NativeHandle::SingleArg(&boolean::native_not), cost_functions::NOT),
+            Equals => NativeFunction("native_eq", NativeHandle::MoreArg(&native_eq), cost_functions::EQ),
+            If => SpecialFunction("native_if", &special_if),
+            Let => SpecialFunction("native_let", &special_let),
+            FetchVar => SpecialFunction("native_var-get", &database::special_fetch_variable),
+            SetVar => SpecialFunction("native_set-var", &database::special_set_variable),
+            Map => SpecialFunction("native_map", &iterables::native_map),
+            Filter => SpecialFunction("native_filter", &iterables::native_filter),
+            Fold => SpecialFunction("native_fold", &iterables::native_fold),
+            Concat => SpecialFunction("native_concat", &iterables::native_concat),
+            AsMaxLen => SpecialFunction("native_as_max_len", &iterables::native_as_max_len),
+            Append => SpecialFunction("native_append", &iterables::native_append),
```

Comment:
Since `special` and `native` are taking 2 distinct paths on cost analysis, should we use this PR as an excuse for doing  a pass on the names in this mapping?  
ex:
`Map => SpecialFunction("native_map", &iterables::native_map),`
should be
`Map => SpecialFunction("special_map", &iterables::special_map), `).

Reply:
Sure

Reply:
What is blocking from just evaluating the first argument in `map` and make it a proper function?

Reply:
@njordhov: Functions are not objects at all in Clarity -- allowing that would make static analysis intractable for things like cost analysis.

Reply:
As functions aren't first class, isn't it the case anyway that only function names would evaluate to functions?